lambda_src/batch_locking_backends/dynamodb.py

```diff
@@ -0,0 +1,127 @@
+'''
+This backend helps lock batches so that they are not processed twice when the API Gateway times out and Snowflake retries the request.
+
+Each batch has a lock that goes through three states:
+
+1.  Un-initialized batch has not yet been seen by GEFF and so a lock does not exist
+2.  Initialized batch is "processing" and has a "locked" lock while the call driver is running
+3.  Finished batch means a call driver has responded or timed out and the lock is "unlocked" and stores the response
+
+In dynamodb, this will correspond to Items like:
+1.  No item
+2.  {"batch_id": "558c5ffb-08a7-4b15-aba7-b7f68edd567f", "locked": true}
+3.  {"batch_id": "558c5ffb-08a7-4b15-aba7-b7f68edd567f", "locked": false, "response": ...}
+'''
+
+import os
+from typing import Dict, Text, List, Any, Tuple, Union
+import boto3
+from json import dumps
+from hashlib import md5
+
+from botocore.exceptions import ClientError
+from ..utils import LOG, ResponseType
+
+AWS_REGION = os.environ.get(
+    'AWS_REGION', 'us-west-2'
+)  # Placeholder while in dev TODO: change as variable/header
+DYNAMODB_TABLE = os.environ.get('DYNAMODB_TABLE_NAME')
+TTL = os.environ.get('DYNAMODB_TABLE_TTL', 86400)
+
+DYNAMODB_RESOURCE = boto3.resource('dynamodb', region_name=AWS_REGION)
```

Comment:
you're only using this in one place -- would it make sense to inline it?

Reply:
Makes sense, inlined it.